File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? 0 : './input.txt';
const input = fs.readFileSync(filePath, 'utf-8').trim().split('\n');

const n = Number(input[0]);
const seq = input[1].split(' ').map(Number);
const x = Number(input[2]);
let left = 0;
let right = n - 1;
let sum = 0;
let cnt = 0;

seq.sort((a, b) => a - b);
while (left < n && left < right) {
  sum = seq[left] + seq[right];
  if (sum === x) {
    cnt += 1;
    right -= 1;
  } else if (sum < x) {
    left += 1;
  } else {
    right -= 1;
  }
}
console.log(cnt);

File: 백준/Gold/9024. 두 수의 합/두 수의 합.py
import sys

input = sys.stdin.readline

t = int(input())

for _ in range(t):
    n, k = map(int, input().split())
    seq = list(map(int, input().split()))
    left = 0
    right = n - 1
    min_diff = float('inf')
    cnt = 0

    seq.sort()
    while left < right:
        t_sum = seq[left] + seq[right]
        t_diff = abs(k - t_sum)
        if t_diff < min_diff:
            min_diff = t_diff
            cnt = 1
        elif t_diff == min_diff:
            cnt += 1
        if t_sum < k:
            left += 1
        else:
            right -= 1
    print(cnt)
